gadget/gadget.go

```diff
@@ -206,6 +208,107 @@ type VolumeUpdate struct {
 	Preserve []string       `yaml:"preserve"`
 }
 
+// DiskVolumeDeviceTraits is a set of traits about a disk that were measured at
+// a previous point in time on the same device, and is used primarily to try and
+// map a volume in the gadget.yaml to a physical device on the system after the
+// initial installation is done. We don't have a steadfast and predictable way
+// to always find the device again, so we need to do a search, trying to find a
+// device which matches each trait in turn, and verify it matches the  physical
+// structure layout and if not move on to using the next trait.
+type DiskVolumeDeviceTraits struct {
+	// each member here is presented in descending order of certainty about the
+	// likelihood of being compatible if a candidate physical device matches the
+	// member. I.e. CachedDevicePath is more trusted than CachedKernelPath is
+	// more trusted than DiskID is more trusted than using the MappedStructures
+
+	// CachedDevicePath is the device path in sysfs and in /dev/disk/by-path the
```

Comment:
Indeed "cached" is not exact, but using something "last" or "previous" might convey some geometrical meaning that would make them even more confusing. More exact could be "[Last]SeenAtDevicePath" but I'm not sure I like it more either :-)